Kosaraju_Algorithm/Kosaraju_Algorithm.py

```diff
@@ -0,0 +1,83 @@
+# Python implementation of Kosaraju's Algorithm 
+# Kosaraju's Algorithm is used to print all Strongly Connected Components (SCCs) in a directed graph
+# A component is said to be strongly connected if there is a path between all pairs of vertices in it.
+
+# Kosaraju’s Algorithm involves two passes of Depth First Search in order to find the number of SCCs
+
+# defaultdict is a useful technique for building graphs.
+from collections import defaultdict 
+
+# Function for first DFS pass and to fill vertices in finishStack
+def firstDFS(g, v, visited, finishStack): 
+	visited[v] = True
+	for i in g[v]: 
+		if visited[i] == False: 
```

Comment:
Too many spaces indention. Please give 4 space indentation at the start of each block.